src/app/core/components/header/header.component.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Component, Input,
+} from '@angular/core';
+import LoginService from 'src/app/auth/service/login.service';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss'],
+})
+class HeaderComponent {
+  constructor(private loginService : LoginService) {}
+
+  isAuthenticated = () : boolean => this.loginService.isAuthenticated();
```

Comment:
as mentioned before - don't use function calls in template, since it's introducing performance issues. Use variable/observables with async